### PR TITLE
Update DM::* functions that return possible errors to use DmResult.  Make bin/test a unit test in dm.rs

### PR DESCRIPTION
**Pull request by trgill**
Update Lower Level API functions to return DmResult.
Remove pub from all most mods (except consts) and explicitly export high level API in lib.rs
Move bin/dmtest.rs to a unit test in dm.rs 
Corrected a couple spelling typos and added some comments

Signed-off-by: Todd Gill <tgill@redhat.com>

### CODE REVIEW

**Comment by trgill** [comment]
@agrover please review.

My intent of this PR is to finish #25 and mostly address #32.

File: src/dm.rs [hunk 1]
```diff
@@ -324,7 +324,7 @@ impl DM {
         };
 
         if new_name.as_bytes().len() > max_len {
-            return Err(Error::new(Other, format!("New name {} too long", new_name)));
+            return Err(DmError::Io(Error::new(Other, format!("New name {} too long", new_name))));
```
Comment (agrover) [inline src/dm.rs]: The old code didn't define its own error type and so kinda crammed everything into an io error. This is one place where maybe we can make a better choice now. How about making this an InternalError?

File: src/dm.rs [hunk 2]
```diff
@@ -744,3 +744,41 @@ impl DM {
...
+    #[test]
+    fn test_basics() {
+        let dmi = DM::new().unwrap();
+
+        println!("Calling version()");
+        let x = dmi.version().unwrap();
+        println!("{:?}", x);
+
+        println!("Calling list_devices()");
+        let x = dmi.list_devices().unwrap();
```
Comment (agrover) [inline src/dm.rs]: This still needs more work but that can happen in a separate PR. This is still a big step in the right direction.

File: src/dm.rs [hunk 3]
```diff
@@ -30,6 +30,7 @@ use types::TargetLine;
 use util::slice_to_null;
 
 /// Major numbers used by DM.
+#[allow(dead_code)]
```
Comment (agrover) [inline src/dm.rs]: yeah I think it can go. We can always resurrect it.

Comment (trgill) [inline src/dm.rs]: Ok

File: src/dm.rs (hunk 1 above)
Comment (trgill) [inline src/dm.rs]: Ok

**Comment by agrover** [comment]
👍

**Pull request merged**